### PR TITLE
Support `device="auto"`for  buffers and set it as default value

### PR DESCRIPTION
**Pull request by qgallouedec**
## Description

See https://github.com/DLR-RM/stable-baselines3/pull/1009

## Context

- [x] I have raised an issue to propose this change ([required](https://github.com/Stable-Baselines-Team/stable-baselines3-contrib/blob/master/CONTRIBUTING.md))

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Checklist:

- [x] I've read the [CONTRIBUTION](https://github.com/Stable-Baselines-Team/stable-baselines3-contrib/blob/master/CONTRIBUTING.md) guide (**required**)
- [ ] The functionality/performance matches that of the source (**required** for new training algorithms or training-related features).
- [ ] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [ ] I have included an example of using the feature (*required for new features*).
- [ ] I have included baseline results (**required** for new training algorithms or training-related features).
- [ ] I have updated the documentation accordingly.
- [x] I have updated the changelog accordingly (**required**).
- [x] I have reformatted the code using `make format` (**required**)
- [x] I have checked the codestyle using `make check-codestyle` and `make lint` (**required**)
- [x] I have ensured `make pytest` and `make type` both pass. (**required**)

Note: we are using a maximum length of 127 characters per line

### CODE REVIEW

**Comment by qgallouedec** [comment]
What do you think about using this PR to do the same for ARS? (See [4f43d6a](https://github.com/Stable-Baselines-Team/stable-baselines3-contrib/pull/98/commits/4f43d6a3e58745349db1bf4a52e5d8936f144e04))

**Comment by araffin** [comment]
> What do you think about using this PR to do the same for ARS? (See [4f43d6a](https://github.com/Stable-Baselines-Team/stable-baselines3-contrib/pull/98/commits/4f43d6a3e58745349db1bf4a52e5d8936f144e04))

ARS is a special case, you should leave it out, cpu should be used in most cases.

File: docs/misc/changelog.rst [hunk 1]
```diff
@@ -24,6 +24,11 @@ Bug Fixes:
 Deprecations:
 ^^^^^^^^^^^^^
 
+Others:
+^^^^^^^
+- Changed the default buffer device from ``"cpu"`` to ``"auto"``
```
Comment (araffin) [inline docs/misc/changelog.rst:29]: I think I need to do a SB3 release if we don't want to break things.

**Review by araffin — APPROVED** [review]
merci =)

**Pull request merged**